UnchainedLauncherCore/src/API/RCON.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace UnchainedLauncher.Core.API
+{
+    public class RCON
+    {
+        protected IPEndPoint RconLocation;
+        public RCON(IPEndPoint rconLocation)
+        {
+            RconLocation = rconLocation;
+        }
+
+        public async Task SendCommand(string command)
+        {
+            await SendCommandTo(RconLocation, command);
+        }
+
+        public static async Task SendCommandTo(IPEndPoint rconLocation, string command)
+        {
+            try
+            {
+                CancellationTokenSource cts = new(1000);
+                using TcpClient client = new TcpClient();
```

Comment:
You can set [SendTimeout](https://learn.microsoft.com/en-us/dotnet/api/system.net.sockets.tcpclient.sendtimeout?view=net-8.0) and [RecieveTimeout](https://learn.microsoft.com/en-us/dotnet/api/system.net.sockets.tcpclient.receivetimeout?view=net-8.0) on the tcp client, which should alleviate the need for the cancellation token